File: 482. License Key Formatting/482. License Key Formatting.py
class Solution(object):
    def licenseKeyFormatting(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        t = s.replace('-', '')
        ist_group_len = len(t) % k or k
        # 25g3j
        res = t[:ist_group_len].upper()
        for i in range(ist_group_len, len(t), k):
            res += '-' + t[i: i + k].upper()

        return res

#"5F3Z-2e-9-w"

print(Solution().licenseKeyFormatting("5F3Z-2e-9-w", 4))

File: 482. License Key Formatting/482. License Key Formatting.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var licenseKeyFormatting = function(s, k) {
    let stringWithDash = s.replace(/-/g, '').toUpperCase();
    let firstGroupLength = stringWithDash.length % k || k;

   let result = stringWithDash.slice(0,firstGroupLength);
   for (let i = firstGroupLength; i < stringWithDash.length; i += k) {
       result += '-' + stringWithDash.slice(i, i + k);
   }
   return result;
};

console.log(licenseKeyFormatting("5F3Z-2e-9-w", 4)); // "5F3Z-2E9W"
console.log(licenseKeyFormatting("2-5g-3-J", 2)); // "2-5G-3J"
console.log(licenseKeyFormatting("--a-a-a-a-a-", 2)); // "AA-AA"